contracts/Governance.sol

```diff
@@ -0,0 +1,256 @@
+        bytes data; // The data (similar to a multisig mechanism).
+        string uriDescription; // URI to a natural language description of the proposal. Also used as ID in mappings.
+        bytes32 descriptionHash; // Hash of the description.
+        string uriArguments; // URI to arguments for the proposal.
+        bytes32 argumentsHash; // Hash of the arguments.
+        ProposalState state; // State of proposal.
+        uint whenPutToVote; // Records the time when a proposal put to vote to be able to calculate voting period.
+        MiniMeToken quorumToken; // The token that will be used for quorum.
+        MiniMeToken voteToken; // The token that will be used for actual proposal voting.
+        bool approved; // Outcome of voting.
+    }
+
+    mapping(bytes32 => Proposal) public proposals;
+
+
+    constructor (uint _proposalQuorum, uint _quorumDivideTime, uint _votingTime, ArbitrablePermissionList _arbitrablePermissionList, Pinakion _pinakion, TokenController _tokenController) public {
+        lastTimeQuorumReached = block.timestamp;
+
+        proposalList = _arbitrablePermissionList;
+        pinakion = _pinakion;
+        tokenController = _tokenController;
+
+        proposalQuorum = _proposalQuorum;
+        currentProposalQuorum = proposalQuorum;
+
+        quorumDivideTime = _quorumDivideTime;
+        votingTime = _votingTime;
+    }
+
+
+    // ****************************** //
+    // *          Modifiers         * //
+    // ****************************** //
+
+    modifier onlyWhenProposalIsNew(bytes32 _id) {
+        require(proposals[_id].state == ProposalState.New, "Only when proposal in New state.");
+        _;
+    }
+
+    modifier onlyWhenProposalPutToSupport(bytes32 _id) {
+        require(proposals[_id].state == ProposalState.PutToSupport, "Only when proposal in PutToSupport state.");
+        _;
+    }
+
+
+    // ****************************** //
+    // *           Events           * //
+    // ****************************** //
+
+    event ProposalCreated(bytes32 indexed _id, address _destination);
+
+    event ProposalRequestedToRegister(bytes32 indexed _id);
+
+    event ProposalPutToSupport(bytes32 indexed _id);
+
+    event ProposalPutToVote(bytes32 indexed _id);
+
+    event ProposalDecided(bytes32 indexed _id, bool _approved);
+
+    event ProposalExecuted(bytes32 indexed _id);
+
+
+
+    // ****************************** //
+    // *    Governance Mechanism    * //
+    // ****************************** //
+
+    /** @dev Creates a proposal, adds to proposals and requests registering to proposalList
+     *  @param _id ID of the proposalList
+     *  @param _destination Destination contract of the execution
+     *  @param _amount Value of the execution
+     *  @param _data Data of the execution
+     *  @param _uriDescription URI of the description of the proposal
+     *  @param _uriArguments URI of the arguments of the proposal
```

Comment:
I would swap the wording in these two names (_uriArguments -> _argumentsURI and _uriDescription -> _descriptionURI)